internal/sftp/goclient.go

```diff
@@ -57,6 +57,32 @@ func (c *GoClient) Upload(ctx context.Context, src io.Reader, dest string) (int6
 	return bytes, nil
 }
 
+// Delete removes the data from dest and returns the
+// number of bytes removed. A new SFTP connection is opened before removing the file, and
+// closed when the delete is complete or cancelled.
+//
+// Delete is not thread safe.
+func (c *GoClient) Delete(ctx context.Context, dest string) (int64, error) {
```

Comment:
I'm realizing that we can't use ctx for cancellation, but I suggest to leave it in case we find a way to use it in the future. I think that both the ssh and sftp packages that we're using predate the context package so they were not designed to accept a context. One alternative that we could consider is to pass the context to the dial method and create our own dialer with [net.DialContext](https://pkg.go.dev/net#Dialer.DialContext), but I don't see it as a blocker.

Reply:
rclone is doing it [here](https://github.com/rclone/rclone/blob/e1b0417c286bdcf5e866d70d51cb75be179c25e6/backend/sftp/ssh_internal.go#L26-L48), but it looks complicated because they support edge cases like proxy settings, etc. It also looks like the ssh package is going to have a DialContext sometime soon: https://go-review.googlesource.com/c/crypto/+/504735.

Reply:
Huh, interesting @sevein.  I thought ctx would be unnecessary for delete because the actual `sftp.Remove()` operation should be very fast, AFAIK.  But now I realize that dialing the server could take a while if there are communications issues, so I agree ctx is still worthwhile for future use.

Reply:
One thing I have considered is doing something similar to what contextio does, or wwe can do @sevein suggestion. I think including context on the actual client struct makes sense and then we can context is equal for operations that normally do not have a context like in contextio